src/loaders/zarrLoader.js

```diff
@@ -103,6 +92,24 @@ export default class ZarrLoader {
     return [data];
   }
 
+  getRasterSize({ z }) {
+    const source = z ? this._data[z] : this._base;
+    const imageHeight = source.shape[this.yIndex];
+    const imageWidth = source.shape[this.xIndex];
+    return { imageHeight, imageWidth };
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  onTileError(err) {
+    // Handle zarr-specific tile Errors
+    // Will check with `err instanceof BoundCheckError` when merged
+    // https://github.com/gzuidhof/zarr.js/issues/47
+    if (!err.message.includes('RangeError')) {
+      // Rethrow error if something other than tile being requested is out of bounds.
+      throw err;
+    }
+  }
```

Comment:
We handle zarr-specific errors. Zarr will perform a check for whether a tile request is "in bounds" so we can `instanceof` that error rather than redundantly checking these bounds. Here, if the error is an instance of this bounding specific error we allow for it (since deck.gl just renders a null tile), and if it's not this error we throw to help with debugging.